### ISSUE TITLE
[Feature Request] GUI Scaling for Hi-DPI displays

### ISSUE THREAD

**Issue by prg318** [DISTRHO/DISTRHO-Ports#25]
Not sure if this is the best place to bring this up, but using some of the ports on 4k or hi-DPI displays can be very difficult (especially something like dexed).  I don't see a way to scale the UI - not sure if this should be handled by juce or the ports themselves..

This would greatly increase usability for a lot of people

**Comment by falkTX** [comment]
hmm I believe they should already work for high-DPI screens.
juce supposedly handles it, but not sure what is the current state on linux.

for vst plugins, the host needs to pass a special attribute reporting the scale. LV2 has no such concept yet.
If hosts don't provide scaling information, I believe juce tries to get it from the system.
some time ago it was only handled in ubuntu, using its gconf settings. not sure how it is now.

**Comment by prg318** [comment]
In my case with Dexed - I can resize the window, but any extra space is just black instead of the dexed GUI reszing.  This also occurs with Obxd.  I will continue to look into this

**Comment by falkTX** [comment]
plugin UIs are not meant to be resizable though, not by default.
the scaling is something different, it makes the UI bigger but it's still non-resizable by the user.

**Comment by a13** [comment]
I have the same problem: dexed/obxd/nekobee controls look tiny on my 4k display

**Comment by falkTX** [comment]
A way to do scale is already present in the update-pugl branch, but there is no auto-detection for it yet.
I know VST2 has a custom property for it, so we can test that (though then the host itself would have to support it too).
For LV2 we will have to define a property ourselves, since one does not exist yet.
DSSI has no way to provide this info in a sorta official way, and I guess we can ignore it.

Which hosts do you know that support scaling?

**Comment by prg318** [comment]
Not positive, but I was under the impression that Bitwig Studio was supposed to support plugin scaling

**Comment by wagenadl** [comment]
Has there been any progress on this issue? The projucer and GUIs that it generates still look tiny in Linux (Ubuntu 18.04) with hidpi screen.

**Comment by hfiguiere** [comment]
Concerning Dexed, I have built the VST from upstream and the resolution is OK. However the one in DISTRHO-Ports is not. Both hosted in Ardour 6-rc1 (happen in 5.12 as well).

Here is the DISTRHO-Ports version:

![Screenshot from 2020-04-30 23-57-34](https://user-images.githubusercontent.com/114441/80781230-a1161180-8b3f-11ea-8789-cac73077f135.png)

And the version from upstream:

![Screenshot from 2020-05-01 00-01-10](https://user-images.githubusercontent.com/114441/80781251-b8ed9580-8b3f-11ea-9cc7-c6d38b59dd9f.png)


Note: the LV2 has the same problem (except the window is properly sized), but I don't have anything to compare it with.

(this is on Fedora 32, and I built both Ardour and the plugins)

**Comment by hfiguiere** [comment]
tl;dr. for Dexed it might be solvable by upgrading to the current upstream code.
